include/netmon_stats.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2023 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdbool.h>
+
+#if defined(CONFIG_NETMON_STATS)
+#include <zephyr/stats/stats.h>
+
+STATS_SECT_START(netmon)
+STATS_SECT_ENTRY32(dhcp_up)
+STATS_SECT_ENTRY32(dhcp_down)
+STATS_SECT_ENTRY32(sntp_time_sync_count)
+STATS_SECT_ENTRY32(sntp_time_req_count)
+STATS_SECT_END;
+
+STATS_NAME_START(netmon)
+STATS_NAME(netmon, dhcp_up)
+STATS_NAME(netmon, dhcp_down)
+STATS_NAME(netmon, sntp_time_sync_count)
+STATS_NAME(netmon, sntp_time_req_count)
+STATS_NAME_END(netmon);
+#else
+struct stats_netmon {
+};
```

Comment:
Where is `struct stats_netmon` defined when `CONFIG_NETMON_STATS=y`? Does the stats subsys create it for us?